packages/next/src/views/NotFound/index.tsx

```diff
@@ -60,7 +60,7 @@ export const NotFoundPage = async ({
     <Fragment>
       <HydrateClientUser permissions={initPageResult.permissions} user={initPageResult.req.user} />
       <DefaultTemplate
-        config={initPageResult.req.payload.config}
+        payload={initPageResult.req.payload}
```

Comment:
This is a breaking change, do we need to do this? Also this is a little misleading to me because `payload` as you have it now _contains_ the config, instead of being the payload class itself, which I believe is how its treated elsewhere.

Reply:
@jacobsfletch The goal of this is to allow the custom `Nav` component to receive `payload` as a prop if it's an RSC - just like any other custom RSC.

`DefaultTemplate` receives `payload` so that it can then pass it to `RenderCustomComponent`, which renders the `CustomNav`.

> Also this is a little misleading to me because payload as you have it now contains the config, instead of being the payload class itself, which I believe is how its treated elsewhere.

Isn't this the same `payload` as any other `payload` which we're passing to RSCs? All of them contains the config

Reply:
**_oic now_**